### PR TITLE
CMR-4314 Add UMM<=>SMAP CollectionDataType mapping and decouple the SMAP expected test from the ISO 19115-2 expected test.

### PR DESCRIPTION
**Pull request by eereiter**

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+(defn split-temporals
+  "Returns a seq of temporal extents with a new extent for each value under key
+  k (e.g. :RangeDateTimes) in each source temporal extent."
+  [k temporal-extents]
+  (reduce (fn [result extent]
+            (if-let [values (get extent k)]
+              (concat result (map #(assoc extent k [%])
+                                  values))
+              (concat result [extent])))
```
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: You could do `(conj result extent)` instead of the concat here.

Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: Although I realize you didn't write this code, just moved it.

File: umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj [hunk 2]
```diff
@@ -232,17 +247,50 @@
          (map cmn/map->DataCenterType)
          distinct)))
 
+(defn- not-provided-begin-date
+  "Returns the default java epoch zero used for not provided date values.
+   This function returns the def record and not just a map so that
+   it can be compared to the actual round trip translation."
+  []
+  (for [x (range 0 1)]
```
Comment (eereiter) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj]: In order to compare the values with the actual I needed the entire defrecord to be in a vector or sequence. Using vec or seq turned every element in the defrecord into a vector and so I couldn't compare the values.  I don't know of any other way to get the result I needed. I tried using vec and seq together, by themselves, in the function, after the function, but I couldn't get the structure I needed.

File: umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj (hunk 1 above)
Comment (eereiter) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: Not sure - I can work on it an try.

File: umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj (hunk 2 above)
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj]: Wrap in brackets: `[(map->RangeDateTimeType {:BeginningDateTime du/parsed-default-date})))]`

File: umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj (hunk 1 above)
Comment (eereiter) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: This produces different results and the tests fail.  Here is the difference between concat and conj 

user=> (concat [1 2 3] [4 5 6])
(1 2 3 4 5 6)

user=> (conj [1 2 3] [4 5 6])
 [1 2 3 [4 5 6]]

Comment (eereiter) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_shared.clj]: Thanks for the help and pairing with me.  I have made the changes.

File: umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj (hunk 2 above)
Comment (eereiter) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj]: Thanks for helping me here too.  I had spacing issues with using the brackets.  I have updated the code.

**Pull request merged**